### PR TITLE
chore(`chisel`): enforce `common::shell` for chisel binary

### PR DESCRIPTION
**Pull request by zerosnacks**
## Motivation

Enforces `common::shell` as added in https://github.com/foundry-rs/foundry/pull/9109 for the `chisel` binary

## Solution

Splits out the enforcement of the common shell per binary / large crate to avoid huge diffs making reviews easier

### CODE REVIEW

**Review by yash-atreya — APPROVED** [review]

**Review by DaniPopes — APPROVED** [review]

File: crates/chisel/bin/main.rs [hunk 1]
```diff
@@ -205,7 +207,7 @@ async fn main_args(args: Chisel) -> eyre::Result<()> {
     }
 
     // Print welcome header
-    println!("Welcome to Chisel! Type `{}` to show available commands.", "!help".green());
+    sh_println!("Welcome to Chisel! Type `{}` to show available commands.", "!help")?;
```
Comment (zerosnacks) [inline crates/chisel/bin/main.rs:210]: cc @DaniPopes in https://github.com/foundry-rs/foundry/pull/9109 you noted messages ought not to be painted in color except for the `warn / error` keyword, is removing this here correct?

Comment (DaniPopes) [reply crates/chisel/bin/main.rs:210]: Yeah but that's for err and warn, and here it's in a REPL too, so it's fine

**Review by grandizzy — APPROVED** [review]
lgtm!

**Comment by zerosnacks** [comment]
Reverted the visual color changes (painting of messages) as this is a special case where users are familiar with the REPL looking a certain way as discussed in https://github.com/foundry-rs/foundry/pull/9177#discussion_r1813163833

**Review by grandizzy — APPROVED** [review]
lgtm!

**Pull request merged**